src/Grid/grid.jl

```diff
@@ -210,14 +215,17 @@ function edges(c::AbstractCell{RefPyramid})
         (ns[2], ns[5]), (ns[4], ns[3]), (ns[3], ns[5]), (ns[4], ns[5]), 
     )
 end
-function faces(c::AbstractCell{RefPyramid})
-    ns = get_node_ids(c)
+function reference_faces(::Type{RefPyramid})
     return (
-        (ns[1], ns[3], ns[4], ns[2]), (ns[1], ns[2], ns[5]), 
-        (ns[1], ns[5], ns[3]), (ns[2], ns[4], ns[5]), 
-        (ns[3], ns[5], ns[4]),                                      
+        (1, 3, 4, 2), (1, 2, 5), 
+        (1, 5, 3), (2, 4, 5), 
+        (3, 5, 4),                                      
     )
 end
+function faces(c::AbstractCell{refshape}) where refshape
+    ns = get_node_ids(c)
+    return ntuple(i->getindex.(Ref(ns), reference_faces(refshape)[i]), nfaces(refshape))
+end
 
```

Comment:
Is there a performance hit to this operation? If so, how severe and can we optimize it?

Reply:
I don't think there's much of a performance hit
The PR:
```julia
julia> @benchmark Ferrite.faces($cell)
BenchmarkTools.Trial: 10000 samples with 1000 evaluations.
 Range (min … max):  2.945 ns … 31.522 ns  ┊ GC (min … max): 0.00% … 0.00%
 Time  (median):     2.956 ns              ┊ GC (median):    0.00%
 Time  (mean ± σ):   2.994 ns ±  0.791 ns  ┊ GC (mean ± σ):  0.00% ± 0.00%

         ▁  ▂ ▂  ▃ ▃  ▅ ▆  █ █  ▇ ▇  ▇ ▇  ▄ ▁                 
  ▂▁▃▁▁▅▁█▁▁█▁█▁▁█▁█▁▁█▁█▁▁█▁█▁▁█▁█▁▁█▁█▁▁█▁█▁▁▇▁▇▁▁▆▁▆▁▁▅▁▂ ▄
  2.94 ns        Histogram: frequency by time        2.97 ns <

 Memory estimate: 0 bytes, allocs estimate: 0.
 ```
 Master:
 ```julia
 julia> @benchmark Ferrite.faces($cell)
BenchmarkTools.Trial: 10000 samples with 1000 evaluations.
 Range (min … max):  2.949 ns … 30.682 ns  ┊ GC (min … max): 0.00% … 0.00%
 Time  (median):     3.024 ns              ┊ GC (median):    0.00%
 Time  (mean ± σ):   3.187 ns ±  1.418 ns  ┊ GC (mean ± σ):  0.00% ± 0.00%

   █▇                                                         
  ▅██▆▆▄▄▄▃▃▆▄▃▇▇▄▄▄▃▃▂▂▂▂▃▃▃▃▃▂▂▂▂▂▂▂▂▂▃▂▁▁▂▁▂▂▁▁▁▁▁▁▁▁▁▂▂▃ ▃
  2.95 ns        Histogram: frequency by time        3.69 ns <

 Memory estimate: 0 bytes, allocs estimate: 0.
 ```